internal/config/config.go

```diff
@@ -0,0 +1,6 @@
+package config
+
+type PartitionNodes struct {
```

Comment:
this was necessary to avoid a cycle

For consistency:  it probably makes sense to move much of cli.Config into here?

Reply:
Or maybe a `partitioning` package for this one?